### DISCUSSION TITLE
Is there a way to enable filtering and pagination for datatables - followup

### DISCUSSION THREAD

**Discussion by plbremer** [squidfunk/mkdocs-material#6597] [Q&A]
Hi, this is a followup to 
https://github.com/squidfunk/mkdocs-material/discussions/5011

I repost it because it appears that posting a reply did not BUMP the topic.

Roughly, like gusmb, I want to enable pagination/filtering

> I am rendering markdown documentation from templates with large datatables. Right now I have enabled the sortable columns feature as explained in the docs. Is there a way to enable filetering on columns and pagination to avoid printing the whole table at once?

squidfunk/gusmb got it sorted out among themselves (by suggesting an approach similar to Tablesort), but I am afraid that I did not get as far. 

For me, 

I have:
1) Updated the extra_javascript portion of mkdocs.yaml to remove conflicts from filter
2) Put what I believe to be the right unpkg link
```
extra_javascript:
  # - https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js
  # - javascripts/tablesort.js
  - https://unpkg.com/tablefilter@0.7.3/dist/tablefilter/tablefilter.js
  - javascripts/tablefilter.js
```

3) Attempted to instantiate a filter class in a way analogous to the way presented in the Tablesort example
```
document$.subscribe(function() {
//   var myTF = new TableFilter(document.querySelector(".my-table"));
  var tables = document.querySelectorAll("article table:not([class])")
  tables.forEach(function(table) {
    new TableFilter(table)
  })
})
```

Of course, "nothing shows up". I suspect that my .js gist is not working. 

I would be grateful for any advice, thanks.

**Comment by kamilkrzyskow** [comment]
Hello @plbremer,
please provide a runnable minimal example, because it's hard for me to understand the input data of your table in Markdown.
https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/

**Comment by plbremer** [comment]
@kamilkrzyskow thanks for taking the time to reply.

What I have, and what is not working, is [9.5.3-plbremer_filter.zip](https://github.com/squidfunk/mkdocs-material/files/13837014/9.5.3-plbremer_filter.zip). 

I was hoping that if I closely paralled the instructions given in https://squidfunk.github.io/mkdocs-material/reference/data-tables/#sortable-tables , as shown in [9.5.3-plbremer_sort.zip](https://github.com/squidfunk/mkdocs-material/files/13837016/9.5.3-plbremer_sort.zip), that I would be successful. 

That is not the case.

**Reply by squidfunk** [reply]
Please post this in #5011 if you want to get help. I've locked conversation here.

**Comment by squidfunk** [comment]
@plbremer: please do not repost. Posting a reply does bump the topic to the top, but you should give people some time to answer. Reposting questions makes our work more tedious because we end up with more duplicates. Closing as duplicate. Thank you.
